integr8ly/integreatly-operator
# MGDAPI-3806 - sts: get token path and role arn from env for functional test

# Issue link

https://issues.redhat.com/browse/MGDAPI-3806

# What

For AWS functional tests, get the role arn and token file path from environment variables for use using sts authentication

# Verification steps



* Provision / Borrow STS cluster
* Create the minimal policies using https://github.com/integr8ly/delorean/pull/261
```
make ocm/sts/rhoam-prerequisites CLUSTER_NAME=<cluster_name> OCM_ENV=<production|staging>
```
* Install RHOAM onto STS cluster using https://github.com/integr8ly/integreatly-operator/pull/2535
* Run the AWS functional test locally
```
ROLE_ARN="arn:aws:iam::408612754352:role/functional_test_role" TOKEN_PATH="/var/run/secrets/openshift/serviceaccount/token" INSTALLATION_TYPE=managed-api TEST="(AWS)|(VPC)" make test/e2e/single
```
* Verify the same by using deleoran and setting the same environment variables
  * Create the following `test-config.yaml` to use built test image
```
---
tests:
  - name: "integreatly-operator-test"
    image: "quay.io/kevfan/integreatly-operator-test-harness:latest"
    timeout: 7200
    envVars:
      - name: "DESTRUCTIVE"
        value: "false"
      - name: "MULTIAZ"
        value: "false"
      - name: "WATCH_NAMESPACE"
        value: "redhat-rhoam-operator"
      - name: "ROLE_ARN"
        value: "arn:aws:iam::408612754352:role/functional_test_role"
      - name: "TOKEN_PATH"
        value: "/var/run/secrets/openshift/serviceaccount/token"

```
  * Run test using in delorean:
```
KUBECONFIG=~/.kube/config DESTRUCTIVE=false MULTIAZ=false ./delorean pipeline product-tests --test-config ./test-config.yaml --output test-results --namespace test-functional | tee testOutput.txt
 ```
